### ISSUE TITLE
Class assignment to ng-dropdown-panel with appendTo does not work with dynamic classes

### ISSUE THREAD

**Issue by cciccole** [ng-select/ng-select#1756]
**Describe the bug**
As described in various places, the class assigned to `ng-dropdown` will be assigned to `ng-dropdown-panel` in the case that `appendTo` is being used.

However, if the class is dynamically assigned that value will not be assigned to the `ng-dropdown-panel`

**Reproducible example**
https://stackblitz.com/edit/angular-7rqm8u?file=src/append-to-example.component.html

**Expected behavior**
The dynamic class is assigned to ng-dropdown-panel.

**Desktop (please complete the following information):**
 - OS: OSX
 - Browser: Chrome

**Related issues**
Here are some related issues but I believe this issue is unique.
- https://github.com/ng-select/ng-select/issues/1199
- https://github.com/ng-select/ng-select/issues/1361

**Comment by varnastadeus** [comment]
#1627

**Comment by yanotoma** [comment]
Hi! I have the same problem but I found a workaround

```
// HTML
<ng-select (focus)="handleFocus()">

// .ts
handleFocus(): void {
   setTimeout(() => {
      const myCustomClass: string ="custom-class"
      const panel = document.querySelector('.ng-dropdown-panel');
      panel.classList.add(myCustomClass);
      console.log('panel', panel);
    }, 0);
}
```

As you can see, when the user focuses the component I look for the panel and then I inject the class

I Hope this can help someone :)

**Comment by vkrupach** [comment]
@yanotoma , thanks for the workaround! Found that "open" handler is better since "focus" does not handle case when an option is selected and then I start typing agian without leaving the field: (open)="this.addDropdownPanelCalss();"

**Comment by yanotoma** [comment]
Yes, you are right! the "open" event is a better approach for this

**Comment by pboutin44** [comment]
Hello I open a PR https://github.com/ng-select/ng-select/pull/2187 which should fix the probleme but hasn't been accepted yet :/
